modules/dnn/src/op_inf_engine.cpp

```diff
@@ -622,6 +622,7 @@ void InfEngineBackendNet::init(int targetId)
 
 #endif  // IE < R5
 
+static Mutex sharedPluginsMutex;
 static std::map<InferenceEngine::TargetDevice, InferenceEngine::InferenceEnginePluginPtr> sharedPlugins;
```

Comment:
It is better to wrap global "static" variables into static functions (to ensure initialization of them at least).

Example:
https://github.com/opencv/opencv/blob/3.4.5/modules/dnn/src/dnn.cpp#L94-L98

Reply:
@alalek, Is that Ok to use a mutex from `getInitializationMutex()`?

https://github.com/opencv/opencv/blob/e82e672a933c8bd713ff53ac9ca553ede169b566/modules/dnn/src/init.cpp#L52-L57

Reply:
I believe you can reuse it (declared in dnn's precomp.hpp).